src/components/modal/Modal.tsx

```diff
@@ -1,43 +1,48 @@
 import React from 'react';
-import { Modal as BaseModal, ModalProps as BaseModelProps, ModalOverrides } from 'baseui/modal';
+import { Modal as BaseModal, ModalProps as BaseModelProps } from 'baseui/modal';
+import { useTheme } from 'hooks';
+import merge from 'deepmerge';
 
 import { border, borderRadius, MountStateNotifier, MountStates } from '../../utils';
 
-const modalOverride = (name?: string): ModalOverrides => ({
-  Dialog: {
-    style: ({ $theme }) => ({
-      ...border($theme.borders.border100),
-      ...borderRadius($theme.borders.radius100),
-    }),
-    props: {
-      'data-testid': `${name ? name + '-' : ''}modal`,
-    },
-  },
-  Close: {
-    style: ({ $theme }) => ({
-      color: $theme.colors.contentTertiary,
-      right: $theme.sizing.scale700,
-      top: $theme.sizing.scale700,
-    }),
-  },
-});
-
 export interface ModalProps extends BaseModelProps {
   children: React.ReactNode;
   onStateChange?: (mountState: MountStates[keyof MountStates]) => void;
   name?: string;
 }
 
-export const Modal = ({ children, onStateChange, name, ...rest }: ModalProps) => {
+export const Modal = ({ children, onStateChange, name, overrides, ...rest }: ModalProps) => {
   // By utilizing the `MountStateNotifier` inside of `BaseModal` it will know the mountstate for when
   // the modal is closed. The base modal is unmounted, but this component `Modal` is still mounted.
   // Hence using a component to check the current state so the modal can use animations for mount states.
+
+  const { theme } = useTheme();
+
+  const defaultOverrides = {
+    Dialog: {
+      style: {
+        ...border(theme.current.borders.border100),
+        ...borderRadius(theme.current.borders.radius100),
+      },
+      props: {
+        'data-testid': `${name ? name + '-' : ''}modal`,
```

Comment:
I noticed that it's not your change, but I noticed a slight mistake:
1. Should be `data-test-id`, not `data-testid`
2. I'd suggest that `data-test-id` should be passed from ModalProps to props as a full value instead of defining name prefixes.

Feel free to react to these suggestions or leave as it is, as it's out of the scope of this PR.

Reply:
It's a design decision. It makes sense to always add a test id. Sometimes you only have 1 modal and than the test id modal is sufficient. Using the name to prefix the test id is also a design decision but as a result the end user no longer has any influence on the test id since there is no override. So if the user wants to use a complete different test id because the current implementation is not aligned with the codebase it's used in then there currently is no way.

About the naming convention. I did a query in the repository and this seems to be the only component with a test id. So there is no standard yet. I believe the reason is since it's part of the initial move. In my opinion, if there are no tests in the current repository it makes no sense to add test ids and it should be something the users passes in as props.